File: python/0053.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        s=[0]*len(nums)
        s[0]=nums[0]
        for i in range(1,len(nums)):
            if s[i-1]>0:
                s[i]=s[i-1]+nums[i]
            else:
                s[i]=nums[i]
        return max(s)

File: rust/src/bin/0053.rs
struct Solution;

use std::cmp::max;

impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut ans = -(1i64 << 32);
        let mut now = 0i64;
        for i in 0..nums.len() {
            if now > 0 {
                now += nums[i] as i64;
            } else {
                now = nums[i] as i64;
            }
            ans = max(ans, now);
        }
        ans as i32
    }
}

fn main() {
    let ans = Solution::max_sub_array(vec![-2, 1, -3, 4, -1, 2, 1, -5, 4]);
    println!("{}", ans);
}